test/pkg/firewall/firewall.go

```diff
@@ -28,23 +32,60 @@ func RunRootCommandOnPod(debugPod *v1.Pod, command string, chrootDir bool, utils
 	return output, nil
 }
 
-func NftListAndWriteToFile(debugPod *v1.Pod, utilsHelpers utils.UtilsInterface, artifactsDir, fileName string) ([]byte, error) {
+func NftListAndWriteToFile(debugPod *v1.Pod, utilsHelpers utils.UtilsInterface, artifactsDir, fileName, tableName, chainName string) (bool, error) {
+	var output []byte
+	var cmdErr error
+
 	command := "nft list ruleset"
-	output, err := RunRootCommandOnPod(debugPod, command, true, utilsHelpers)
-	if err != nil {
-		return nil, fmt.Errorf("failed to list NFT ruleset on node %s: %w", debugPod.Spec.NodeName, err)
+	gomega.Eventually(func() bool {
+		output, cmdErr = RunRootCommandOnPod(debugPod, command, true, utilsHelpers)
+		if cmdErr != nil {
+			log.Printf("Error listing nft ruleset: %v", cmdErr)
+			return false
+		}
+
+		if len(output) == 0 {
+			log.Println("No nft rules found in output.")
+			return false
+		}
+
+		if !strings.Contains(string(output), tableName) || !strings.Contains(string(output), chainName) {
+			log.Printf("Required table %s or chain %s not found in nftables output.", tableName, chainName)
+			return false
+		}
+
+		return true
+	}, 2*time.Minute, 10*time.Second).Should(gomega.BeTrue(), "Timeout while waiting for nftables to contain table %s and chain %s", tableName, chainName)
+
+	if cmdErr != nil {
```

Comment:
This is checked in the Eventually, if it fails there it will timeout and fail and won't get here